frontend/eda/event-streams/EventStreamPage/EventStreamDetails.tsx

```diff
@@ -0,0 +1,110 @@
+  const { data: eventStream } = useGetItem<EdaEventStream>(edaAPI`/event-streams/`, params.id);
+  const getPageUrl = useGetPageUrl();
+  const restartPolicyHelpBlock = (
+    <>
+      <p>{t('A policy to decide when to restart a rulebook.')}</p>
+      <br />
+      <p>{t('Policies:')}</p>
+      <p>{t('Always: restarts when a rulebook finishes.')}</p>
+      <p>{t('Never: never restarts a rulebook when it finishes.')}</p>
+      <p>{t('On failure: only restarts when it fails.')}</p>
+    </>
+  );
+  if (!eventStream) {
+    return <LoadingPage />;
+  }
+  return (
+    <Scrollable>
+      <PageDetails>
+        <PageDetail label={t('Event stream ID')}>{eventStream?.id || ''}</PageDetail>
+        <PageDetail label={t('Name')}>{eventStream?.name || ''}</PageDetail>
+        <PageDetail label={t('Description')}>{eventStream?.description || ''}</PageDetail>
+        <PageDetail label={t('Source type')}>{eventStream?.source_type || ''}</PageDetail>
+        {eventStream.credentials && eventStream.credentials.length > 0 && (
+          <PageDetail label={t('Credential(s)')}>
+            <LabelGroup>
+              {eventStream.credentials.map((credential) => (
+                <Label key={credential?.id}>{credential?.name}</Label>
+              ))}
+            </LabelGroup>
+          </PageDetail>
+        )}
+        <PageDetail
+          label={t('Decision environment')}
+          helpText={t('Decision environments are a container image to run Ansible rulebooks.')}
+        >
+          {eventStream && eventStream?.decision_environment?.id ? (
+            <Link
+              to={getPageUrl(EdaRoute.DecisionEnvironmentPage, {
+                params: { id: eventStream?.decision_environment?.id },
+              })}
+            >
+              {eventStream?.decision_environment?.name}
+            </Link>
+          ) : (
+            eventStream?.decision_environment?.name || ''
+          )}
+        </PageDetail>
+        <PageDetail label={t('Event stream status')}>
+          <StatusCell status={eventStream?.status || ''} />
+        </PageDetail>
+        <PageDetail label={t('Restart policy')} helpText={restartPolicyHelpBlock}>
+          {eventStream?.restart_policy ? restartPolicyName(eventStream?.restart_policy, t) : ''}
+        </PageDetail>
+        <PageDetail label={t('Created')}>
+          {eventStream?.created_at ? formatDateString(eventStream?.created_at) : ''}
+        </PageDetail>
+        <LastModifiedPageDetail
+          value={eventStream?.modified_at ? formatDateString(eventStream?.modified_at) : ''}
+        />
+      </PageDetails>
+      {eventStream?.source_args && (
+        <PageSection variant="light">
+          <PageDetailCodeEditor
+            value={eventStream.source_args}
+            showCopyToClipboard={true}
+            label={t('Arguments')}
+            helpText={t('Arguments')}
+          />
+        </PageSection>
+      )}
+    </Scrollable>
+  );
```

Comment:
can we have data-cys added to the above elements here?

Reply:
data-cy added by the components used